pkg/xen-tools/initrd/chroot2.c

```diff
@@ -24,60 +26,53 @@ struct clone_args {
 #define STACK_SIZE (8 * 1024 * 1024)
 static char child_stack[STACK_SIZE];    /* Space for child's stack */
 
-static int childFunc(void *args)
+static int child_func(void *args)
 {
-    clone_args *parsed_args = (clone_args *)args;
-
+    struct clone_args *parsed_args = args;
     struct passwd *pws;
+
     pws = getpwuid(parsed_args->uid);
-    if (pws == NULL) {
-        perror("getpwuid() failed:");
-        return -1;
-    }
+    if (pws == NULL)
+        err(-1, "getpwuid(%d) failed:", parsed_args->uid);
```

Comment:
_err()_ is going to exit the program, so you should use a positive value as return value here. You can use EXIT_FAILURE (defined at stdlib.h) or just 1....

Reply:
Why? I want 255 exit status.